app/subscriber/src/features/settings/MyAccountSettings.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { FaEnvelope, FaToggleOff, FaToggleOn, FaUmbrellaBeach } from 'react-icons/fa';
+import { toast } from 'react-toastify';
+import { useUsers } from 'store/hooks';
+import { useProfileStore } from 'store/slices';
+import { ISubscriberUserModel, ToggleButton } from 'tno-core';
+
+import * as styled from './styled';
+
+const MyAccountSettings = () => {
+  const { updateUser } = useUsers();
+  const [{ profile, impersonate }, { storeMyProfile, storeImpersonate }] = useProfileStore();
+  const isVacationMode: boolean = !!impersonate
+    ? impersonate?.preferences?.isVacationMode ?? false
+    : profile?.preferences?.isVacationMode ?? false;
+
+  const toggleVacationMode = React.useCallback(async () => {
+    if (!profile) {
+      toast.error('User information is missing. Please try again later');
+      return;
+    }
+    const baseProfile = impersonate ?? profile;
+    const createUser = (): ISubscriberUserModel => {
+      // use impersonate if it exists, otherwise use profile
+      return {
+        ...baseProfile,
+        preferences: {
+          ...baseProfile.preferences,
+          isVacationMode: !isVacationMode,
+        },
+      };
+    };
+    const user = createUser();
+
+    try {
+      !!impersonate ? storeImpersonate(user) : storeMyProfile(user);
```

Comment:
This logic is already performed by the `useUser` hook.  I don't see a need to do it again.